### DISCUSSION TITLE
Sonoff D1 dimmer

### DISCUSSION THREAD

**Discussion by mrmin24** [arendst/Tasmota#11299] [Q&A] [answered]
Hi,

I have tested the D1 dimmer with the Itead firmware as well as ESPHome and Tasmota and cannot seem to get the dimming to work properly. It seems that the dimmer is trying to dim, but the level barely changes. 

I tested on Dimmable LED and even on normal LED and CFL just to see what happens. On the dimmable LED I see no change, on the other two they start to struggle to stay on stable, but they do not flicker as what I would expect. 

Is there any way to possibly change the dimming levels to test this?  Or any other steps I am missing?

I use 230V/50Hz. 

Thx

**Comment by Jason2866** [comment, marked as answer]
Try with a incandescent light bulb. If this is working your dimmable LED are not compatible.
In general the Sonoff D1 dimmer is not that great...

**Reply by mrmin24** [reply]
Thx, was planning to do this test as soon as I can find an incandescent, they are rare. 

I get the feeling that the dimming should just be scaled up, but I am unsure how much control is possible. I will dig some more

**Reply by mrmin24** [reply]
I have tried with Incandescent and Halogen and they both work perfectly. I have also tried with the filament type LEDs and they do work, but has some flicker. 

So for the moment at least it does not seem that Dimmable LEDs work at least not all of them. Strangely enough when I connect a dimmable LED in parallel with one of the others that work (mentioned above) then the LED does dim somewhat, but not all the way down. 

I tested with ESPHome

**Comment by gr6803** [comment]
I got the D1 working with normal bulbs. The dimming behaviour was ok but the dimmer switches randomly on sometimes. So I consider it to be more a toy than a device :-)

**Reply by sfromis** [reply]
Note that support for Sonoff D1 is still in "experimental" status, not completed to "full satisfaction".
There is a long thread about the battles with Sonoff D1, for "inspiration" :-) 
https://github.com/arendst/Tasmota/issues/7598
https://templates.blakadder.com/sonoff_D1.html

**Reply by mrmin24** [reply]
The thread mostly talks about the RF433 part and not about the actual dimming. Nobody seems to complain about the dimming as such. Also watched a couple of videos of demos. 

Will try some different bulbs.

**Reply by sfromis** [reply]
You specifically wrote "but the dimmer switches randomly on sometimes" which is not about dimming as such, but about random switching events, which IIRC is also a topic in the linked thread. That's why I posted it.
